components/tidb_query_datatype/src/codec/batch/lazy_column_vec.rs

```diff
@@ -145,17 +145,29 @@ impl LazyBatchColumnVec {
     /// Encode into chunk format.
     // TODO: Move to other place.
     pub fn encode_chunk(
-        &mut self,
+        self,
         logical_rows: &[usize],
         output_offsets: &[u32],
         schema: &[FieldType],
         output: &mut Vec<u8>,
         ctx: &mut EvalContext,
     ) -> Result<()> {
-        for offset in output_offsets {
-            let offset = *offset as usize;
-            let col = &self.columns[offset];
-            col.encode_chunk(ctx, logical_rows, &schema[offset], output)?;
+        if !output_offsets.is_empty() {
+            let maximum_offset = *output_offsets.last().unwrap() as usize + 1;
```

Comment:
The `maximum_offset` may be incorrect, since `output_offsets` is not ensured to be ordered. Also there may be duplicates (*in very rare cases*) in `output_offsets`.

Reply:
This part is now reverted to original code. We may not be able to eliminate this copy. Let's see whether directly copy data may lead to performance improvement.